Feature/get involved

jotalis opened the pull request:
Built out the involved section features including:
- Header text
- Speech bubbles with application buttons / links
- Anteater SVGs in background
- Box Responsiveness

Todo:
- Attach application links to buttons

Nice to Haves:
- Framer animations on anteater SVGs
- Speech bubble hooks if needed
- Text Responsiveness if needed

jotalis committed:
feat: add get involved section header, speech bubbles, anteaters, and responsiveness

jotalis committed:
refactor: rename Mentor section to Involved

jotalis committed:
fix: change z-index property to allow mentor link to be clicked

waalbert commented on apps/site/src/app/(home)/sections/Involved/Involved.module.scss:
```diff
@@ -0,0 +1,125 @@
+$primary-color: #78638A;
+$secondary-color: #DAFFF5;
+$white: #fff;
+$black: #000;
```
Amazing work! The text color and border color aren't exactly #000, so please refer to the figma for accurate colors. Also, please use the colors found in _zothacks-theme.scss if possible. Otherwise, you can the colors declared here

waalbert commented on apps/site/src/app/(home)/sections/Involved/Involved.module.scss:
```diff
@@ -0,0 +1,125 @@
+@use "zothacks-theme" as theme;
```
Change indent size to 4 spaces

jotalis committed:
fix: use theme colors and change indent to 4 spaces

jotalis committed:
feat: added shake and scale animations on anteater svgs

jotalis committed:
fix tweak shadow color

jotalis committed:
chore: merge with main:

jotalis committed:
chore: remove old color definitions

jotalis commented:
Preview:

<img width="1187" alt="{8B9DC8C0-7EA1-4528-99E0-E228D6EB1CA8}" src="https://github.com/user-attachments/assets/69c5c056-505b-4ac5-a8c0-c83a3d5ab56d">
<img width="579" alt="{D82F4E2B-7DA2-4DFD-9540-724A4AD6803C}" src="https://github.com/user-attachments/assets/1484e8ba-5cec-4edd-bdb1-22bed40f8e1b">
<img width="430" alt="{856BF4CB-C997-4017-9F24-F5689626AC69}" src="https://github.com/user-attachments/assets/6c7a4e6f-0f1a-4359-a0b0-eacb3006f741">



Fixed / Added:
- SCSS tab spacing
- Fixed colors to Figma accurate colors using theme variables
- Animations on anteater svgs

taesungh commented on apps/site/src/assets/images/involved_anteater_left.svg:
Chore [same for other image]: embedding a raster in a vector is inefficient when no other vector capabilities are utilized, i.e. this image could be a regular PNG file, or even better would be the actual vectorized source.

taesungh commented on apps/site/src/app/(home)/sections/Involved/Involved.tsx:
```diff
@@ -0,0 +1,79 @@
+		</Button>
+	);
+
+	const hoverAnimation = {
+		scale: 1.05,
+		rotate: [0, 10, -10, 0],
+	};
+
+	const hoverTransition = {
+		duration: 0.5,
+		ease: "easeInOut",
+	};
```
Question: is there a particular reason you decided to implement these visual effects with Framer? The motion here should be relatively simple to implement with native [CSS animations](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_animations/Using_CSS_animations).

jotalis replied:
we were already using framer somewhere else, so it was convenient, but i've gone ahead and converted them to native CSS animations

taesungh commented on apps/site/src/app/(home)/sections/Involved/Involved.tsx:
```diff
@@ -0,0 +1,79 @@
+import { motion } from "framer-motion";
+
+const Mentor = () => {
+	const sectionHeader = <h2 className={styles.headerText}>GET INVOLVED</h2>;
```
Nitpick: these elements are headings, not headers

taesungh commented on apps/site/src/app/(home)/sections/Involved/Involved.tsx:
```diff
@@ -0,0 +1,79 @@
+					<motion.img
+						className={styles.anteaterLeft}
+						src={anteater_left.src}
+						alt="Involved Anteater Left"
```
Chore: this alt text does not provide a meaningful description of the image. Consider something like "A cartoon anteater sitting on a ring buoy".

taesungh commented on apps/site/src/app/(home)/sections/Involved/Involved.module.scss:
```diff
@@ -0,0 +1,125 @@
+$speech-padding: 20px; 
+$speech-shadow: 0 10px 0 rgba(1, 57, 70, 0.2);
+
+@mixin speechBubble($bg-color, $border-color) {
```
Praise: nice job using a mixin here.

taesungh commented on apps/site/src/app/(home)/sections/Involved/Involved.tsx:
```diff
@@ -0,0 +1,79 @@
+	);
+};
+
+export default Mentor;
```
Chore: for consistency, component name should match the file name. Also, naming it just `Involved` seems a bit awkward, consider `GetInvolved`?

taesungh commented on apps/site/src/app/(home)/sections/Involved/Involved.module.scss:
```diff
@@ -0,0 +1,125 @@
+	}
+}
+
+.applyButton::before {
```
Suggestion: this selector could be nested with the previous.

jotalis committed:
fix: use png over raster svg

jotalis committed:
chore: rename header to heading

jotalis committed:
chore: change framer animations to native css animations and improve image alt text

jotalis committed:
fix: remove blue border from buttons

jotalis committed:
chore: nest button styles

waalbert commented on apps/site/src/app/(home)/sections/GetInvolved/GetInvolved.tsx:
```diff
@@ -0,0 +1,69 @@
+
+import styles from "./GetInvolved.module.scss";
+
+import { motion } from "framer-motion";
```
This can be removed since you're not using framer anymore

waalbert commented on apps/site/src/app/(home)/sections/GetInvolved/GetInvolved.tsx:
```diff
@@ -0,0 +1,69 @@
+		<p>Otherwise, if you have some experience under your belt,</p>
+	);
+	const applyLink = (
+		<Button href="#" type="button" className={styles.applyButton}>
```
Can you change the href to /apply?

waalbert commented on apps/site/src/app/(home)/sections/GetInvolved/GetInvolved.tsx:
```diff
@@ -0,0 +1,69 @@
+	);
+
+	const mentorLink = (
+		<Button href="#" type="button" className={styles.applyButton}>
```
Can you change the href to /mentor?

jotalis committed:
chore: change link hrefs and remove framer import

waalbert committed:
merge from main

waalbert committed:
shift container up

waalbert committed:
clean up css
